packages/@aws-cdk/aws-ec2/lib/client-vpn-endpoint-types.ts

```diff
@@ -0,0 +1,57 @@
+import { IDependable, IResource } from '@aws-cdk/core';
+import { IConnectable } from './connections';
+
+/**
+ * A client VPN endpoint
+ */
+export interface IClientVpnEndpoint extends IResource, IConnectable {
+  /**
+   * The endpoint ID
+   */
+  readonly endpointId: string;
+
+  /**
+   * Dependable that can be depended upon to force target networks associations
+   */
+  readonly targetNetworksAssociated: IDependable;
+}
+
+/**
+ * A Lambda function
+ */
+export interface IFunction {
```

Comment:
This only works in TypeScript, not in nominally-typed languages.

What you'd normally have to do is define an interface (or abstract class) for integrations and make a concrete class in a separate package combining EC2 types and Lambda types.

However in this case, I believe `aws-lambda` already depends on `aws-ec2`. So it's permissible in this case to define an interface here and have `aws-lambda.Function` implement it directly.

Reply:
This is now `IClientVpnConnectionHandler`, and `FunctionBase` implements it.